File: samples/python/preview.py
"""
License: Apache 2.0. See LICENSE file in root directory.
Copyright(c) 2020-2021 Intel Corporation. All Rights Reserved.
"""
import time
import os
import rsid_py


def on_image(frame):
    print(f'got_frame #{frame.number} {frame.width}x{frame.height}')
    if frame.number > 10:
        os._exit(0)


if __name__ == '__main__':
    preview_cfg = rsid_py.PreviewConfig()
    preview_cfg.camera_number = 0
    preview_cfg.device_type = rsid_py.DeviceType.F45x  # or rsid_py.DeviceType.F46x or use  rsid_py.discover_device_type(PORT)
    p = rsid_py.Preview(preview_cfg)
    p.start(on_image, snapshot_callback=None)
    while True: time.sleep(10)

File: samples/c/preview.c
// License: Apache 2.0. See LICENSE file in root directory.
// Copyright(c) 2020-2021 Intel Corporation. All Rights Reserved.

#include "rsid_c/rsid_preview.h"
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#ifdef LINUX
#include <unistd.h>
#else
#include <Windows.h>
#endif

void render(rsid_image image, void* ctx)
{
    printf("frame #%d: %dX%d (%d bytes)\n", image.number, image.width, image.height, image.size);
}

int main()
{
    rsid_preview_config config;
    rsid_preview* preview;
    config.camera_number = -1; // auto detect
    config.preview_mode = MJPEG_1080P;

    preview = rsid_create_preview(&config);
    rsid_start_preview(preview, render, NULL);

    printf("run preview for 30 sec\n");
#ifdef LINUX
    sleep(30); // seconds
#else
    Sleep(30000); // miliseconds
#endif
    rsid_stop_preview(preview);
}
